stablehlo/dialect/TypeInference.cpp

```diff
@@ -3555,13 +3551,12 @@ LogicalResult verifyDynamicIotaOp(std::optional<Location> location,
                                   Value outputShape, int64_t iotaDimension,
                                   Value result) {
   auto shape = result.getType().cast<ShapedType>();
```

Comment:
We should consider moving these casts to be `RankedTensorType`. This will make it very explicit that `hasRank` would be true. Let's do this in a followup if we decide to, since it will apply to most ops eventually.